File: Shows/ShowPHP/Show/VehicleFactory.php
<?php

namespace TheRayCode\Factory;

class VehicleFactory
{
    public function createVehicle($type) {
        if ($type == 'car') {
            return new Car();
        } elseif ($type == 'motorcycle') {
            return new Motorcycle();
        } elseif ($type == 'bicycle') {
            return new Bicycle();
        } else {
            throw new InvalidArgumentException("Invalid vehicle type");
        }
    }
}

File: Csharp/Creational/Factory/Example/VehicleFactory.cs
using System;

namespace Show
{
    public class VehicleFactory
    {
        public static IVehicle GetVehicle(string vehicleType)
        {
            switch (vehicleType)
            {
                case "Car":
                    return new Car();
                case "Motorcycle":
                    return new Motorcycle();
                default:
                    throw new ArgumentException("Invalid vehicle type");
            }
        }
    }
}